src/classes/UTIL_Where.cls

```diff
@@ -0,0 +1,765 @@
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Utilities
+* @description Utility used to build the WHERE clause
+*/
+public class UTIL_Where {
+    /** @description Exception class */
+    public class CriteriaException extends Exception {}
+
+    /** @description Filter operator options */
+    public Enum Operator {
+        EQUALS,
+        NOT_EQUALS,
+        IN_SET
+    }
+    
+    /*********************************************************************************************************
+    * @description Translates field expression operator to a String
+    * @param operant The operator
+    * @return String The string value of the operator
+    */
+    private static String operatorToString(Operator operant) {
+        String result = null;
+
+        if (operant == Operator.EQUALS) { result = '='; }
+        else if (operant == Operator.NOT_EQUALS) { result = '!='; }
+        else if (operant == Operator.IN_SET) { result = 'IN'; }
+
+        return result;
+    }
+
+    /*********************************************************************************************************
+    * @description Determines if the criteria is met for the match results and the operator
+    * @param matchResult The field expression(s) match results
+    * @param logicOperator The logic ooperator (AND/OR)
+    * @return Boolean
+    */
+    public static Boolean meetsCriteria(List<Boolean> matchResult, String logicOperator) {
+        if (matchResult == null || matchResult.isEmpty()) {
+            return true;
+        }
+
+        if (logicOperator == AND_OPERATOR) {
+            for (Boolean result : matchResult) {
+                if (!result) {
+                    return false;
+                }
+            }
+            return true;
+
+        } else {
+            for (Boolean result : matchResult) {
+                if (result) {
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
+    
+    /** @description AND logic operator */
+    public static final String AND_OPERATOR = 'AND';
```

Comment:
I see the `AND_OPERATOR` constant defined twice - once here and once in `UTIL_Query`' Is it worth combining the operator constants into a single location?

Reply:
This would create dependency between `UTIL_Finder` and `UTIL_Query` as well as introduce more testing of the code where `UTIL_Query` has been used. It is better to leave them as they are for now unless further changes require it.